### PR TITLE
DYFI Charts improvement and Responses tab

### LINKED ISSUE #1013: DYFI Charts
Intensity vs. Distance
Responses vs. Time

Try using ngx-charts, but fall back to D3 if it's not sophisticated enough for the intensity vs. distance plot

### LINKED ISSUE #1014: DYFI Responses
Sortable table with sorting mechanism

Show 10 by default

Separate download button

### LINKED ISSUE #1040: Improve dyfi chart code
dyfi service
- does this need to be outside the dyfi module?

intensity vs distance
- the bubbleSeries and lineSeries variables should be cleared if clearing dyfiSeries.
- dyfiSeries appears to be unused
- checking `value === null || !value` could be simplified to `!value` (null is falsy)
- can a local `series` variable be used instead of modifying the `dyfiData` object?

bubble-series
- can the values be preconverted to numbers?
- can the `isActive` check be moved outside the `map` function
- it would be clearer if you handle the undefined case early then filter to remove null values:
```
if (!d.y || !d.x) {
  return null
}
// do stuff
```

### PR DESCRIPTION
**Pull request by dslosky-usgs**
closes #1040 
closes #1013 
closes #1014

Moved DYFI service into DYFI module
Fixed some awkward coding (it came from ngx-charts I swear!!!)

# Intensity Vs. Distance
- Implemented log axis for intensity vs. distance chart
- Added fixed axis functionality and will follow up with @vinceq-usgs if it's alright that fixing the axis will hide some data
- Added x-y grid

# Responses vs. time
- Investigated mapping artifacts... These might actually be caused by the way the library zooms... By zooming into an area and running your mouse along the line, you will actually see the lines readjust... interesting...

# Responses tab
Initial table rendering and sorting and paginating using a Material table

# Other

Still having some trouble stopping the intensity vs. distance plot from growing its container... Responses vs. time renders just fine until the other plot grows the container

We should make separate issues for the remaining items, including:
Intensity vs. distance plot size (mentioned above)
Graphing artifacts at high zoom on responses vs. time chart
Switching to a scatter plot at a specific zoom level for the responses vs. time
Adding UTM Block and city/location name when available in the data
Adding a fixed-size sliding scale on the x-axis for the intensity vs. distance plot

# Other other

This PR includes an additional scripts input ...
```
"scripts": [
+              {
+                "input": "node_modules/document-register-element/build/document-register-element.js"
+              },
               {
                 "input": "node_modules/document-register-element/build/document-register-element.js"
               }
...
```

Which I think will fix the issue I found on beta, #1047. Super weird, but with only a single `"input"`, the web component fails to render...  Thoughts?

### CODE REVIEW

File: src/app/dyfi/responses/responses.component.html [hunk 1]
```diff
@@ -1,3 +1,73 @@
-<p>
-  responses works!
-</p>
+<ng-container *ngIf="loaded">
+  <mat-table [dataSource]="responses" matSort>
```
Comment (ehunter-usgs) [inline src/app/dyfi/responses/responses.component.html]: For the default, do we want to sort by distance?
```
<mat-table [dataSource]="responses" matSort matSortActive="dist" matSortDirection="asc">
```

File: src/app/shared/ngx-charts/bubble-chart/bubble-series/bubble-series.component.ts [hunk 2]
```diff
@@ -39,66 +39,80 @@ export class BubbleSeriesComponent extends SwimlaneBubbleSeries {
   getCircles(): any[] {
     const seriesName = this.data.name;
 
+    const isActive = this.activeEntries && !this.activeEntries.length ? true : (this.isActive({name: seriesName}));
+
     return this.data.series.map((d, i) => {
-      if (typeof d.y !== 'undefined' && typeof d.x !== 'undefined') {
```
Comment (ehunter-usgs) [inline src/app/shared/ngx-charts/bubble-chart/bubble-series/bubble-series.component.ts]: I think you want to keep this logic `typeof d.y !== 'undefined'` if you say `!d.y` when d.y = 0, then null will be returned. I'm guessing that zero is a valid value?

File: src/app/shared/ngx-charts/bubble-line-chart/bubble-line-chart/bubble-line-chart.component.ts [hunk 3]
```diff
@@ -213,51 +240,57 @@ export class BubbleLineChartComponent extends BaseChartComponent  {
...
+            values.push(d.max);
           }
         }
       }
     }
 
-    let min = Math.min(...domain);
-    const max = Math.max(...domain);
+    let min, max;
+    if (this.yScaleMin && !this.autoScale) {
```
Comment (ehunter-usgs) [inline src/app/shared/ngx-charts/bubble-line-chart/bubble-line-chart/bubble-line-chart.component.ts]: Why change how min and max are calculated?

File: src/app/dyfi/responses/responses.component.html (hunk 1 above)
Comment (dslosky-usgs) [reply src/app/dyfi/responses/responses.component.html]: Good call!

File: src/app/shared/ngx-charts/bubble-chart/bubble-series/bubble-series.component.ts (hunk 2 above)
Comment (dslosky-usgs) [reply src/app/shared/ngx-charts/bubble-chart/bubble-series/bubble-series.component.ts]: Ahh, that's a good call Eddie

File: src/app/shared/ngx-charts/bubble-line-chart/bubble-line-chart/bubble-line-chart.component.ts (hunk 3 above)
Comment (dslosky-usgs) [reply src/app/shared/ngx-charts/bubble-line-chart/bubble-line-chart/bubble-line-chart.component.ts]: This way allows us to set a strict y-axis min/max; the "default" method I grabbed from the ngx-charts example code actually ignored those values

**Pull request merged**